src/usermgmt.rs

```diff
@@ -27,23 +30,32 @@ fn create_kdf(pepper: &[u8]) -> Argon2 {
     Argon2::new_with_secret(pepper, Argon2id, V0x13, params).expect("failed to initialize Argon2")
 }
 
-struct Session(String);
+#[derive(Serialize, Debug, sqlx::FromRow)]
+#[serde(rename_all = "camelCase")]
+pub struct AccountSession {
+    pub id: i64,
+    email: String,
+    #[serde(skip_serializing)]
+    session_id: Vec<u8>,
+}
 
-impl Session {
-    async fn create(uid: i64, db: &DB) -> eyre::Result<Self> {
+impl AccountSession {
+    async fn create(id: i64, email: String, db: &DB) -> eyre::Result<Self> {
         let mut session_id = [0u8; SESSION_ID_BYTES];
-        let mut inserted = false;
         for _ in 0..3 {
             OsRng.fill_bytes(&mut session_id);
             let insert_result = sqlx::query("insert into session (id, account_id) values ($1, $2)")
                 .bind(&session_id[..])
-                .bind(uid)
+                .bind(id)
                 .execute(db)
                 .await;
             match insert_result {
                 Ok(_) => {
-                    inserted = true;
-                    break;
+                    return Ok(Self {
+                        id,
+                        email,
+                        session_id: session_id.to_vec(),
```

Comment:
Is there a reason for converting to Vec here?

Reply:
My own inexperience with spelling this type differently in a way that satisfies sqlx and the compiler, mostly :p

`[u8]` in `SessionAccount` makes the struct unsized, but `Sized` is needed for `sqlx::FromRow`.

`[u8; SESSION_ID_BYTES]` seems unsuitable for both `.bind()` and `query_as` calls, but maybe there's an extra trait I need to import?

`&'a [u8]` requires a lifetime everywhere... Think it would likely need a copy equivalent of `to_vec()` anyway if `session_id` here is on the stack?

Decided `Vec<u8>` was easiest, but if you have suggestions or alternatives I'm all ears.

Reply:
Ah, right, FromRow...

I guess this is the place where it's easiest to formulate my dissatisfaction with this whole thing: `AccountSession` combines two unrelated things (account information and session id), and so it falls under the limitations of both.

Reply:
I can certainly refactor more if you'd prefer me to lump it into this MR, it's just more code. Like I said I wasn't fully happy with it either, but disagree that they're "unrelated" -- they're clearly related -- but I can understand the desire to keep the types as normalized as the db schema.

Not sure what you mean by "limitations" of both. That session id's `[u8; SESSION_ID_BYTES]` type isn't ergonomic, which is why you originally converted it to a base64 encoded `String` as early as possible to begin with?